addon/-private/create-reducer-helper.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember';
+
+const {
+  Helper,
+  K,
+  computed,
+  observer,
+  set,
+  get
+} = Ember;
+
+/**
+ * Creates a generic Helper class implementation that expects a `needle` and
+ * `haystack` as arguments. A `reducer` function is required to be passed in
+ * that reduces the `haystack` down into a single value.
```

Comment:
It's a good point, I was struggling to come up with a name for it because of how generic it is...